utils/server/schemaPages.ts

```diff
@@ -0,0 +1,76 @@
+import { GetServerSidePropsContext } from "next";
+
+import prisma from "utils/server/prisma";
+import { getCachedSession } from "utils/server/session";
+import { SchemaPageHeaderProps } from "components/SchemaPageFrame/SchemaPageFrame";
+
+export const getSchemaPageHeaderData = async (
```

Comment:
I'm not really sure what's important to optimize and what isn't, and I see the value in factoring out the common logic in a utility function here, but this does result in a lot of database requests that all have to wait for each other in series - plus some of it is not going to get used (ie the version count isn't used on the version detail page) and some of it is redundant (we'll end up re-fetching more information about the current version on the main page).

Maybe a cleaner way to deduplicate some of the logic is to instead export some common query fragments from here? Prisma should make it pretty easy to export typed re-usable fragments like

```typescript
export const findSchema = (profileSlug: string, contentSlug: string): SchemaWhereInput => ({
	slug: contentSlug,
	agent: {
		OR: [{ user: { slug: profileSlug } }, { organization: { slug: profileSlug } }],
	},
})
```

and then we still build a single query in each `getServerSideProps`

```typescript
const schema = await prisma.schema.findFirst({
	where: findSchema(profileSlug, contentSlug),
	select: { ...everything-this-specific-page-needs }
})
```